### ISSUE TITLE
nextProps aren't handled by no-unused-props

### ISSUE THREAD

**Issue by remcohaszing** [yannickcr/eslint-plugin-react#1079]
I have the following (simplified) piece of code:

```js
  componentWillReceiveProps(nextProps) {
    if (!nextProps.error) {
      console.log('oh noez');
    }
  }
```

This triggers the following ESLint error:

```
  89:27  error  'error' is missing in props validation  react/prop-types
```

However, if I do add `error` to `propTypes`, another error is triggered:

```js
  33:10  error  'error' PropType is defined but prop is never used  react/no-unused-prop-types
```

I think the two rules should cooperate.

I'm using the following relevant packages:

- eslint@3.15.0
- eslint-config-airbnb@14.1.0
- eslint-plugin-react@6.10.0

**Comment by emirdeliz** [comment]
+1

**Comment by ljharb** [comment]
It's not really about "cooperation", but certainly both rules should be able to detect used props in `nextProps` identically.
